applications/FluidDynamicsApplication/custom_elements/symbolic_stokes.cpp

```diff
@@ -0,0 +1,4943 @@
+    IndexType NewId, GeometryType::Pointer pGeometry)
+    : FluidElement<TElementData>(NewId, pGeometry) {}
+
+template <class TElementData>
+SymbolicStokes<TElementData>::SymbolicStokes(
+    IndexType NewId, GeometryType::Pointer pGeometry, Properties::Pointer pProperties)
+    : FluidElement<TElementData>(NewId, pGeometry, pProperties) {}
+
+template <class TElementData>
+SymbolicStokes<TElementData>::~SymbolicStokes() {}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Public Operations
+
+template <class TElementData>
+Element::Pointer SymbolicStokes<TElementData>::Create(
+    IndexType NewId,
+    NodesArrayType const &ThisNodes,
+    Properties::Pointer pProperties) const
+{
+    return Kratos::make_intrusive<SymbolicStokes>(NewId, this->GetGeometry().Create(ThisNodes), pProperties);
+}
+
+template <class TElementData>
+Element::Pointer SymbolicStokes<TElementData>::Create(
+    IndexType NewId,
+    GeometryType::Pointer pGeom,
+    Properties::Pointer pProperties) const
+{
+    return Kratos::make_intrusive<SymbolicStokes>(NewId, pGeom, pProperties);
+}
+
+template <class TElementData>
+void SymbolicStokes<TElementData>::CalculateLocalSystem(
+    MatrixType &rLeftHandSideMatrix,
+    VectorType &rRightHandSideVector,
+    ProcessInfo &rCurrentProcessInfo)
+{
+    // Resize and intialize output
+    if (rLeftHandSideMatrix.size1() != LocalSize)
+        rLeftHandSideMatrix.resize(LocalSize, LocalSize, false);
+
+    if (rRightHandSideVector.size() != LocalSize)
+        rRightHandSideVector.resize(LocalSize, false);
+
+    noalias(rLeftHandSideMatrix) = ZeroMatrix(LocalSize, LocalSize);
+    noalias(rRightHandSideVector) = ZeroVector(LocalSize);
+
+    if (TElementData::ElementManagesTimeIntegration){
+        TElementData data;
+        data.Initialize(*this, rCurrentProcessInfo);
+
+        //Get Shape function data
+        Vector gauss_weights;
+        Matrix shape_functions;
+        ShapeFunctionDerivativesArrayType shape_derivatives;
+        this->CalculateGeometryData(gauss_weights, shape_functions, shape_derivatives);
+        const unsigned int number_of_gauss_points = gauss_weights.size();
+        // Iterate over integration points to evaluate local contribution
+        for (unsigned int g = 0; g < number_of_gauss_points; g++) {
+            UpdateIntegrationPointData(data, g, gauss_weights[g], row(shape_functions, g), shape_derivatives[g]);
+            this->AddTimeIntegratedSystem(data, rLeftHandSideMatrix, rRightHandSideVector);
+        }
+
+    } else{
+        KRATOS_ERROR << "SymbolicStokes is supposed to manage time integration." << std::endl;
+    }
+}
+
+template <class TElementData>
+void SymbolicStokes<TElementData>::CalculateRightHandSide(
+    VectorType &rRightHandSideVector,
+    ProcessInfo &rCurrentProcessInfo)
+{
+    MatrixType tmp;
+    CalculateLocalSystem(tmp, rRightHandSideVector, rCurrentProcessInfo);
+}

```

Comment:
This is an overhead. The `CalculateRightHandSide` can be easily implemented with the automatic differentiation.

Reply:
Now calling the base CalculateRightHandSide. This should avoid the problem of overhead.

Reply:
Now calling the base `CalculateRightHandSide` . This should resolve the problem of overhead.